src/prepare-beta/package.ts

```diff
@@ -0,0 +1,56 @@
+// prepare-beta/package.ts
+
+import path from 'node:path';
+import { readFile, writeFile } from 'node:fs/promises';
+import { debug } from 'debug';
+
+import { getPRNumber } from './github';
+
+const log = debug('publish-beta:package');
+
+interface PackageJSON {
+  name: string;
+  version: string;
+  files: string[];
+}
+
+const NUMBER_OF_CHARS_TO_USE_FROM_COMMIT_SHA = 4;
+
+export function generatePackageBetaTag(): string {
+  const commentSha = process.env['GITHUB_SHA'];
+  if (!commentSha) {
+    throw new Error('Unable to get GITHUB_SHA');
+  }
+  const id = commentSha.slice(-NUMBER_OF_CHARS_TO_USE_FROM_COMMIT_SHA, commentSha.length);
+  const prNumber = getPRNumber();
+  return `PR.${prNumber}-${id}`;
+}
+
+function checkFilesPropertyExists(packageJSON: string): void {
```

Comment:
I'm not even sure we need this, since this is a beta publish why not give the flexibility of allowing the default set of files.

Reply:
a lot of packages don't have a files property so this was made to force migration. I'll remove it from the new process and leave it on the old beta publish?